src/main/java/Duke.java

```diff
@@ -1,10 +1,226 @@
+import exceptions.InvalidCommandException;
+import exceptions.UnknownCommandException;
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
+
 public class Duke {
+    // Assume no more than 100 tasks
+    private static final int MAX_TASKS = 100;
+    private static final ArrayList<Task> tasks = new ArrayList<>(MAX_TASKS);
+    private static final String DIRECTORY_NAME = "data";
+    private static final String FILE_NAME = "Duke.txt";
+
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        loadTasks();
+        Scanner in = new Scanner(System.in);
+        greet();
+        while (executeCommand(in.nextLine()));
```

Comment:
Consider using more descriptive variable names
Perhaps `scanner` would be better.
This is apparent when you read the while loop as `in.nextLine()` doesn't really make sense linguistically.

Reply:
Thanks for pointing that out.